aptos-move/framework/src/natives/cryptography/algebra/gas.rs

```diff
@@ -3,5 +3,166 @@
 // Copyright (c) Aptos
 // SPDX-License-Identifier: Apache-2.0
 
+use crate::natives::cryptography::algebra::{HashToStructureSuite, Structure};
+use move_core_types::gas_algebra::{InternalGas, InternalGasPerArg, NumArgs};
+
+fn log2_floor(n: usize) -> usize {
```

Comment:
Can we use the pre-existing `log2_floor` code in `aptos-move/framework/src/natives/cryptography/ristretto255_point.rs` which has test cases and guards for `n == 0`?

We can move the code there & the `log2_ceil` code below into `aptos-move/framework/src/natives/helpers.rs`.

We also need to add test cases for `log2_ceil`, as we have for `log2_floor`.

Reply:
Fixed.